pkg/provider/aws/resources/vpc.go

```diff
@@ -103,84 +105,107 @@ func CreateNetwork(config *config.Application, dag *core.ResourceGraph) *Vpc {
 	if dag.GetResource(vpc.Id()) != nil {
 		return vpc
 	}
-
 	region := NewRegion()
-	azs := NewAvailabilityZones()
-	igw := NewInternetGateway(appName, "igw1", vpc)
+	dag.AddDependency(vpc, region)
+
 	publicRt := &RouteTable{
 		Name: fmt.Sprintf("%s-public", vpc.Name),
 		Vpc:  vpc,
-		Routes: []*RouteTableRoute{
-			{CidrBlock: "0.0.0.0/0", GatewayId: core.IaCValue{Resource: igw, Property: ID_IAC_VALUE}},
-		},
 	}
-	dag.AddDependenciesReflect(publicRt)
 
-	dag.AddResource(region)
-	dag.AddDependency(azs, region)
-	dag.AddResource(vpc)
-	dag.AddDependency(vpc, region)
-	dag.AddResource(igw)
-	dag.AddDependency(igw, vpc)
+	importId := config.Imports[vpc.Id()]
+	if importId == "" {
+		igw := NewInternetGateway(appName, "igw1", vpc)
+		dag.AddDependenciesReflect(igw)
 
-	az1 := core.IaCValue{
-		Resource: azs,
-		Property: "0",
+		publicRt.Routes = append(publicRt.Routes, &RouteTableRoute{CidrBlock: "0.0.0.0/0", GatewayId: core.IaCValue{Resource: igw, Property: ID_IAC_VALUE}})
 	}
-	az2 := core.IaCValue{
-		Resource: azs,
-		Property: "1",
+	dag.AddDependenciesReflect(publicRt)
+
+	var importSubnetIds []core.ResourceId
+	for id := range config.Imports {
+		if id.Provider == AWS_PROVIDER && strings.HasPrefix(id.Type, VPC_SUBNET_TYPE_PREFIX) {
+			importSubnetIds = append(importSubnetIds, id)
+		}
 	}
 
-	public1 := CreatePublicSubnet("public1", az1, vpc, "10.0.128.0/18", dag)
+	if len(importSubnetIds) > 0 {
+		for _, id := range importSubnetIds {
+			sn := &Subnet{
+				Name: id.Name,
+				Vpc:  vpc,
+			}
+			err := sn.SetTypeFromId(id)
```

Comment:
In the templates_compiler `renderResource` it does the following:
```go
	deps := tc.resourceGraph.GetDownstreamResources(resource)
	for _, dep := range deps {
		imp, ok := dep.(*imports.Imported)
		if ok {
			return tc.renderResourceImport(out, resource, imp, tmpl)
		}
	}
```

So it will replace the rendering of the imported resource.